### PR TITLE
only wait once per disconnect for new connection

### PR DESCRIPTION
**Pull request by mmaxim**
As discussed on chat, this makes it so that each "source" only waits for a new connection once per disconnect. This will make it so that we don't keep waiting 4 seconds after we have established we are offline and no new connection is likely coming.

### CODE REVIEW

File: go/chat/sourceofflinable.go [hunk 1]
```diff
@@ -55,20 +56,26 @@ func (s *sourceOfflinable) IsOffline(ctx context.Context) bool {
...
+				return s.offline
+			case <-time.After(4 * time.Second):
+				s.Lock()
+				defer s.Unlock()
+				s.delayed = true
+				s.Debug(ctx, "IsOffline: timed out")
+				return s.offline
+			}
+		} else {
+			s.Debug(ctx, "IsOffline: offline, but skipping delay since we already did it")
```
Comment (patrickxb) [inline go/chat/sourceofflinable.go]: i think you still need a `return s.offline`?

Comment (mmaxim) [reply go/chat/sourceofflinable.go]: It just falls though to return it

Comment (patrickxb) [reply go/chat/sourceofflinable.go]: Ok, github diff fail on my part then...

**Comment by mmaxim** [comment]
I'll try this out on a custom build before deploying to master to make sure it is ok.

**Pull request merged**